bootstrap

```diff
@@ -56,28 +66,33 @@ fetch_and_compile({Name, ErlFirstFiles}, Deps) ->
     ok = fetch(Repo, Name),
     compile(Name, ErlFirstFiles).
 
-fetch({git, Url, Source}, App) ->
+fetch({pkg, Name, Vsn}, App) ->
     Dir = filename:join([filename:absname("_build/default/lib/"), App]),
-    case filelib:is_dir(Dir) of
-        true ->
-            true = code:add_path(filename:join(Dir, "ebin")),
-            ok;
-        false ->
-            fetch_source(Dir, Url, Source),
-            ok
+    CDN = "https://s3.amazonaws.com/s3.hex.pm/tarballs",
+    Package = binary_to_list(<<Name/binary, "-", Vsn/binary, ".tar">>),
+    Url = string:join([CDN, Package], "/"),
+    case request(Url) of
+        {ok, Binary} ->
+            {ok, Contents} = extract(Binary),
+            ok = erl_tar:extract({binary, Contents}, [{cwd, Dir}, compressed]);
+        _ ->
+            io:format("Error: Unable to fetch package ~p ~p~n", [Name, Vsn])
     end.
 
-fetch_source(Dir, Url, {ref, Ref}) ->
-    ok = filelib:ensure_dir(Dir),
-    os:cmd(io_lib:format("git clone ~s ~s", [Url, Dir])),
-    {ok, Cwd} = file:get_cwd(),
-    file:set_cwd(Dir),
-    os:cmd(io_lib:format("git checkout -q ~s", [Ref])),
-    file:set_cwd(Cwd);
-fetch_source(Dir, Url, {_, Branch}) ->
-    ok = filelib:ensure_dir(Dir),
-    os:cmd(io_lib:format("git clone ~s ~s -b ~s --single-branch",
-                         [Url, Dir, Branch])).
+extract(Binary) ->
+    {ok, Files} = erl_tar:extract({binary, Binary}, [memory]),
+    {"contents.tar.gz", Contents} = lists:keyfind("contents.tar.gz", 1, Files),
```

Comment:
Meh, I don't think so. If you have a copy of rebar3 (which you can dl from rebar3.org) you can already build a new copy with verified packages.